### PR TITLE
requirements.txt: bring it up to date 

### PR DESCRIPTION
**Pull request by XhmikosR**
Not sure why those deps where there, but it seems everything works locally without those using Python 3.7.6

```
C:\Users\xmr\Desktop\docs>pip install -r requirements.txt
Collecting markdown-include==0.5.1
  Using cached https://files.pythonhosted.org/packages/ef/44/eb6e9b4fa1110b719abb876c9b6dd8b46af886a94536ec4e9117fe5e7b97/markdown-include-0.5.1.tar.gz
Collecting mkdocs==1.0.4
  Using cached https://files.pythonhosted.org/packages/db/f9/b0179afee0db21943120ea606eb68bda1257b96420df74b775280eb5850b/mkdocs-1.0.4-py2.py3-none-any.whl
Collecting mkdocs-material==4.6.0
  Using cached https://files.pythonhosted.org/packages/b1/e6/2d6457a421f72993ce6468d4c20458a8b8431e3a92b4126a588defd469e4/mkdocs_material-4.6.0-py2.py3-none-any.whl
Collecting markdown
  Using cached https://files.pythonhosted.org/packages/c0/4e/fd492e91abdc2d2fcb70ef453064d980688762079397f779758e055f6575/Markdown-3.1.1-py2.py3-none-any.whl
Collecting livereload>=2.5.1
  Using cached https://files.pythonhosted.org/packages/12/4d/30cfe74402d2e962d66d35da29bf8850b0557b559ce84d09967c8ade859e/livereload-2.6.1-py2.py3-none-any.whl
Collecting PyYAML>=3.10
  Using cached https://files.pythonhosted.org/packages/9e/9c/d4865f9b24c7cfe83181e892ec5ade1435cde46bc606bb5ac2b297d75c38/PyYAML-5.3-cp37-cp37m-win_amd64.whl
Collecting Jinja2>=2.7.1
  Using cached https://files.pythonhosted.org/packages/65/e0/eb35e762802015cab1ccee04e8a277b03f1d8e53da3ec3106882ec42558b/Jinja2-2.10.3-py2.py3-none-any.whl
Collecting click>=3.3
  Using cached https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl
Collecting tornado>=5.0
  Using cached https://files.pythonhosted.org/packages/4e/f8/86ab848da5c79715f901e22bc23d2dc467b8672fe1eda8d3e1f48cba130c/tornado-6.0.3-cp37-cp37m-win_amd64.whl
Collecting mkdocs-minify-plugin>=0.2
  Using cached https://files.pythonhosted.org/packages/66/fe/47e0acd1df33a289a075af252f3d7d65fb8acec425f5fac12493669c6828/mkdocs-minify-plugin-0.2.1.tar.gz
Collecting pymdown-extensions<6.3,>=6.2
  Using cached https://files.pythonhosted.org/packages/be/c5/6187d810dff4458783a9d714b5131fd4d2156e680ba17012dd5bc7e97954/pymdown_extensions-6.2.1-py2.py3-none-any.whl
Collecting Pygments>=2.2
  Using cached https://files.pythonhosted.org/packages/be/39/32da3184734730c0e4d3fa3b2b5872104668ad6dc1b5a73d8e477e5fe967/Pygments-2.5.2-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=36 in c:\users\xmr\appdata\local\programs\python\python37\lib\site-packages (from markdown->markdown-include==0.5.1->-r requirements.txt (line 1)) (44.0.0)
Collecting six
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting MarkupSafe>=0.23
  Using cached https://files.pythonhosted.org/packages/65/c6/2399700d236d1dd681af8aebff1725558cddfd6e43d7a5184a675f4711f5/MarkupSafe-1.1.1-cp37-cp37m-win_amd64.whl
Collecting htmlmin>=0.1.4
  Using cached https://files.pythonhosted.org/packages/b3/e7/fcd59e12169de19f0131ff2812077f964c6b960e7c09804d30a7bf2ab461/htmlmin-0.1.12.tar.gz
Collecting jsmin>=2.2.2
  Using cached https://files.pythonhosted.org/packages/17/73/615d1267a82ed26cd7c124108c3c61169d8e40c36d393883eaee3a561852/jsmin-2.2.2.tar.gz
Collecting pep562
  Using cached https://files.pythonhosted.org/packages/f5/31/2c8475b148c8a0c8cae39300f4346068e21909d44e0659f104ce76c3a7ef/pep562-1.0-py2.py3-none-any.whl
Installing collected packages: markdown, markdown-include, tornado, six, livereload, PyYAML, MarkupSafe, Jinja2, click, mkdocs, htmlmin, jsmin, mkdocs-minify-plugin, pep562, pymdown-extensions, Pygments, mkdocs-material
    Running setup.py install for markdown-include ... done
    Running setup.py install for htmlmin ... done
    Running setup.py install for jsmin ... done
    Running setup.py install for mkdocs-minify-plugin ... done
Successfully installed Jinja2-2.10.3 MarkupSafe-1.1.1 PyYAML-5.3 Pygments-2.5.2 click-7.0 htmlmin-0.1.12 jsmin-2.2.2 livereload-2.6.1 markdown-3.1.1 markdown-include-0.5.1 mkdocs-1.0.4 mkdocs-material-4.6.0 mkdocs-minify-plugin-0.2.1 pep562-1.0 pymdown-extensions-6.2.1 six-1.13.0 tornado-6.0.3
```

### CODE REVIEW

**Comment by XhmikosR** [comment]
Unless I'm misunderstanding the whole idea of requirements.txt. Is it supposed to list all needed packages including sub-packages? If so, it needs to be updated still.

**Comment by dschaper** [comment]
It needs to be generated by `pip freeze` and yes it should be a verbose listing of everything.

**Comment by XhmikosR** [comment]
I thought so too after doing some reading. Please check one more time.

File: requirements.txt [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 Jinja2==2.10.3
+jsmin==2.2.2
 livereload==2.6.1
 Markdown==3.1.1
 markdown-include==0.5.1
 MarkupSafe==1.1.1
 mkdocs==1.0.4
 mkdocs-material==4.6.0
+mkdocs-minify-plugin==0.2.1
+pep562==1.0
```
Comment (XhmikosR) [inline requirements.txt]: I'm not sure if this is needed.

**Comment by XhmikosR** [comment]
Actually @dschaper what's the exact command one has to use? I see packages that are not directly related to the docs repo, like backports-abc, pep562 and singledispatch

**Comment by XhmikosR** [comment]
I think now the patch should be OK:

```
C:\Users\xmr\Desktop\docs>pip install mkdocs mkdocs-material markdown-include
Collecting mkdocs
  Using cached mkdocs-1.0.4-py2.py3-none-any.whl (1.2 MB)
Collecting mkdocs-material
  Using cached mkdocs_material-4.6.0-py2.py3-none-any.whl (722 kB)
Collecting markdown-include
  Using cached markdown-include-0.5.1.tar.gz (4.0 kB)
Collecting PyYAML>=3.10
  Using cached PyYAML-5.3-cp37-cp37m-win_amd64.whl (215 kB)
Collecting Jinja2>=2.7.1
  Using cached Jinja2-2.10.3-py2.py3-none-any.whl (125 kB)
Collecting click>=3.3
  Using cached Click-7.0-py2.py3-none-any.whl (81 kB)
Collecting tornado>=5.0
  Using cached tornado-6.0.3-cp37-cp37m-win_amd64.whl (415 kB)
Collecting Markdown>=2.3.1
  Using cached Markdown-3.1.1-py2.py3-none-any.whl (87 kB)
Collecting livereload>=2.5.1
  Using cached livereload-2.6.1-py2.py3-none-any.whl (23 kB)
Collecting pymdown-extensions<6.3,>=6.2
  Using cached pymdown_extensions-6.2.1-py2.py3-none-any.whl (198 kB)
Collecting mkdocs-minify-plugin>=0.2
  Using cached mkdocs-minify-plugin-0.2.1.tar.gz (2.7 kB)
Collecting Pygments>=2.2
  Using cached Pygments-2.5.2-py2.py3-none-any.whl (896 kB)
Collecting MarkupSafe>=0.23
  Using cached MarkupSafe-1.1.1-cp37-cp37m-win_amd64.whl (16 kB)
Requirement already satisfied: setuptools>=36 in c:\users\xmr\appdata\local\programs\python\python37\lib\site-packages (from Markdown>=2.3.1->mkdocs) (45.1.0)
Collecting six
  Using cached six-1.14.0-py2.py3-none-any.whl (10 kB)
Collecting pep562
  Using cached pep562-1.0-py2.py3-none-any.whl (7.0 kB)
Collecting htmlmin>=0.1.4
  Using cached htmlmin-0.1.12.tar.gz (19 kB)
Collecting jsmin>=2.2.2
  Using cached jsmin-2.2.2.tar.gz (12 kB)
Installing collected packages: PyYAML, MarkupSafe, Jinja2, click, tornado, Markdown, six, livereload, mkdocs, pep562, pymdown-extensions, htmlmin, jsmin, mkdocs-minify-plugin, Pygments, mkdocs-material, markdown-include
    Running setup.py install for htmlmin ... done
    Running setup.py install for jsmin ... done
    Running setup.py install for mkdocs-minify-plugin ... done
    Running setup.py install for markdown-include ... done
Successfully installed Jinja2-2.10.3 Markdown-3.1.1 MarkupSafe-1.1.1 PyYAML-5.3 Pygments-2.5.2 click-7.0 htmlmin-0.1.12 jsmin-2.2.2 livereload-2.6.1 markdown-include-0.5.1 mkdocs-1.0.4 mkdocs-material-4.6.0 mkdocs-minify-plugin-0.2.1 pep562-1.0 pymdown-extensions-6.2.1 six-1.14.0 tornado-6.0.3
```

I did this after removing all my global packages.

**Comment by dschaper** [comment]
> Actually @dschaper what's the exact command one has to use? I see packages that are not directly related to the docs repo, like backports-abc, pep562 and singledispatch

I use `pip freeze`, see https://websauna.org/docs/tutorials/deployment/freeze.html for docs on that.

**Comment by dschaper** [comment]
It's the documentation site so if the merge borks then we can just revert it.

**Comment by XhmikosR** [comment]
I think everything is fine now. If we want to be safe just purge Travis CI cache and restart the job. If it's green it should be fine :)

**Pull request merged**